sources/game/Game.gd

```diff
@@ -111,3 +125,21 @@ func puzzle_quit() -> void:
 	puzzle.queue_free()
 
 	puzzle = null
+
+
+"""Handle pause menu events
+"""
+func on_pause() -> void:
+	if level:
+		level.get_tree().paused = true
+	
+func on_resume() -> void:
+	if level:
+		level.get_tree().paused = true
+	pass
+
+func on_go_to_main_menu() -> void:
```

Comment:
Please, don't do that, do not mix multiple type in the same Array based on arbitrary knowledge, its a nightmare to debug such things and feels unnatural. Use Utils.main_menu() instead.

Reply:
Thanks. I misunderstood about levels in Game.gd. Will fix it